packages/block-editor/src/components/font-appearance-control/index.js

```diff
@@ -116,6 +117,11 @@ export default function FontAppearanceControl( props ) {
 					( weight ) => weight.value === face.fontWeight
 				)
 			) {
+				// Check if font weight includes a space, if so it must be a variable font.
+				if ( /\s/.test( face.fontWeight ) ) {
```

Comment:
We should probably trim left and right, split by space, and try to get the first and second numbers.

Reply:
But in the case of Inter, doesn't the font weight string `100 900` represent a range of weights from 100-900? So in this case, it would only list 100 and 900 in the editor.

Maybe we should split by space and then try to work out the list of available fonts between the two numbers? e.g. in the case of `100 900`, we can assume we want all the weights between 100-900: 100, 200, 300, 400, 500, 600, 700, 800, and 900.

Reply:
I think the tricky part is that CSS Fonts level 4 supports weight 1000, and some Google web fonts support that weight too. For example, it seems like there are fonts with a weight 1000 in [this list of variable fonts](https://fonts.google.com/variablefonts) as well.

Also, I don't know much about variable fonts, but is it possible that the weights are not in units of 100? If so, it's too complicated for this PR and we might be able to keep the logic as it is now.

Reply:
I agree, I think for now, at least the behaviour in this PR is better than trunk, and we can follow-up with more detailed handling for variable fonts.

Reply:
I think I wasn't clear enough. I reformulate: the presence of a space character is not a great way to tell if a font face is variable. 

Look at this example. Notice the trailing space in the fontWeight of the first fontFace. The space breaks the implementation, and the result is listing all the rage of font weights. Probably we should apply `trim()` before checking for spaces.
```
{
					"fontFace": [
						{
							"fontFamily": "Cardo",
							"fontStyle": "normal",
							"fontWeight": "400 ",
							"src": [
								"file:./assets/fonts/cardo/cardo_normal_400.woff2"
							]
						},
						{
							"fontFamily": "Cardo",
							"fontStyle": "italic",
							"fontWeight": "400",
							"src": [
								"file:./assets/fonts/cardo/cardo_italic_400.woff2"
							]
						},
					],
					"fontFamily": "Cardo",
					"name": "Cardo",
					"slug": "heading"
				}
```

Reply:
Ah I see! Moved the `.trim()` before we check for spaces here: https://github.com/WordPress/gutenberg/pull/61915/commits/aba764dc91a26685b3a4f51420f26b94ed42f935.

Reply:
@mikachan I think what we had before https://github.com/WordPress/gutenberg/commit/aba764dc91a26685b3a4f51420f26b94ed42f935 was actually what we need. `String.trim()` doesn't modify the value in place; it returns a new value.

Reply:
Thanks @creativecoder - I've tested @matiasbenedetto's example above with both doing `.trim()` first separately, and having it within the if statement, and both work the same. So I think the version with less code is best (i.e. before https://github.com/WordPress/gutenberg/commit/aba764dc91a26685b3a4f51420f26b94ed42f935). Thanks! Updated here: https://github.com/WordPress/gutenberg/pull/61915/commits/7bd07ab08120159cd65df8087605687bd36a439a. Also improved the inline comment.